storey/aggregations.py

```diff
@@ -415,67 +418,115 @@ def __getitem__(self, key):
 
 
 class AggregationBuckets:
-    def __init__(self, name, aggregation, window, base_time, max_value, is_hidden=False, initial_data=None):
+    def __init__(self, name, aggregation, base_time, max_value, is_hidden, initial_windows, initial_data=None):
         self.name = name
         self.aggregation = aggregation
-        self.window = window
         self.max_value = max_value
         self.buckets = []
-        self.is_hidden = is_hidden
         self.should_persist = True
         self.pending_aggr = {}
         self.storage_specific_cache = {}
+        self.base_time = base_time
+        self.explicit_windows = None
+        self.hidden_windows = None
+        self._initial_data = initial_data
+        self.last_bucket_start_time = None
+        self.first_bucket_start_time = None
+
+        if is_hidden:
+            self.hidden_windows = initial_windows
+        else:
+            self.explicit_windows = initial_windows
+
+        self.period_millis = initial_windows.period_millis
```

Comment:
Why do you need a copy of - `self.period_millis, self.max_window_millis, self.total_number_of_buckets`
you can just have like a Getter method to `get_window_periods()`, and `get_max_window` that will take the max of both hidden and explicit, etc..